rgbd/src/server_rgbd.cpp

```diff
@@ -0,0 +1,127 @@
+#include "rgbd/server_rgbd.h"
+
+#include <opencv2/core/mat.hpp>
+#include <opencv2/imgcodecs.hpp>
+
+#include <rgbd_msgs/RGBD.h>
+
+#include <ros/node_handle.h>
+
+#include <tue/serialization/conversions.h>
+
+#include "rgbd/image.h"
+#include "rgbd/serialization.h"
+
+namespace rgbd {
+
+const int ServerRGBD::MESSAGE_VERSION = 2;
+
+// ----------------------------------------------------------------------------------------
+
+ServerRGBD::ServerRGBD()
+{
+}
+
+// ----------------------------------------------------------------------------------------
+
+ServerRGBD::~ServerRGBD()
+{
+    nh_.shutdown();
+    send_thread_.join();
+    service_thread_.join();
+}
+
+// ----------------------------------------------------------------------------------------
+
+void ServerRGBD::initialize(const std::string& name, RGBStorageType rgb_type, DepthStorageType depth_type, const float service_freq)
+{
+    pub_image_ = nh_.advertise<rgbd_msgs::RGBD>(name, 1);
+    rgb_type_ = rgb_type;
+    depth_type_ = depth_type;
+
+    nh_.setCallbackQueue(&cb_queue_);
+    service_server_ = nh_.advertiseService(name, &ServerRGBD::serviceCallback, this);
+    service_thread_ = boost::thread(&ServerRGBD::serviceThreadFunc, this, service_freq);
+}
+
+// ----------------------------------------------------------------------------------------
+
+void ServerRGBD::send(const Image& image, bool threaded)
+{
+    threaded = false; // Running threaded causes a memory leak.
```

Comment:
This doesn't make any sense!

Reply:
#hack

Reply:
I have removed the threading as it isn't used and only caused troubles.